File: codility/3.TimeComplexity/TapeEquilibrium/Solution.java
class Solution {
    public int solution(int[] A) {
        // write your code in Java SE 8
        int result  = 100000;
        int partSum = 0;
        int allSum = 0;
        
        //sum all the array elements 
        for (int i = 0; i < A.length; i++) {
            allSum += A[i];  
        }
        
        //add elements except the last element 
        for (int i = 0; i < A.length-1; i++) {
            partSum += A[i];
            allSum -= A[i];
            result = Math.min(result, Math.abs(partSum - allSum));
        }
        return result;
    }
}

File: codility/3.TimeComplexity/TapeEquilibrium/solution.py
class Solution:
  def solution(self, A):
    """
    :type A: List[int]
    :rtype: int
    """
    result = 100000
    n = len(A)
    psum = 0
    #sum all the elements
    asum = sum(A[i] for i in range(0, n))
    
    for i in range(n-1):
      psum += A[i]
      asum -= A[i]
      result = min(result, abs(asum-psum))
    return result